sycl/test/invoke_simd/return-type-uniform.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o /dev/null
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/oneapi/experimental/invoke_simd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl;
+namespace esimd = sycl::ext::intel::esimd;
+
+[[intel::device_indirectly_callable]] uniform<int>
+callee(simd<int, 8>) SYCL_ESIMD_FUNCTION {
+  return uniform<int>(5);
+}
```

Comment:
Using 'uniform' in ESIMD context does not seem needed at all (IMO). "int callee(simd< int, 8 > vec, int scalar) SYCL_ESIMD_FUNCTION" is enough to express the uniform-ness of return value and the second argument.

I suppose this patch is needed because of this invoke_simd SPEC statements that implicitly tells using uniform in ESIMD context is Ok, right?
> Return values of type T are converted to sycl::ext::oneapi::experimental::uniform< T >, and broadcast to each work-item; every work-item in the sub-group receives the same value.

Reply:
Yes, this is specifically needed because the `invoke_simd` spec defines `uniform` as a valid return type for ESIMD function thats are called through `invoke_simd`. We don't have a way to check at compile time if an ESIMD function is called by invoke_simd because it is/can be through a function pointer, so we just allow it in all cases.